torchrl/envs/transforms/transforms.py

```diff
@@ -4109,74 +4109,158 @@ class RewardSum(Transform):
+            value will prevail over the environment ``reset_keys``.
 
+    Examples:
+        >>> from torchrl.envs.transforms import RewardSum, TransformedEnv
+        >>> from torchrl.envs.libs.gym import GymEnv
+        >>> env = TransformedEnv(GymEnv("Pendulum-v1"), RewardSum())
+        >>> td = env.reset()
+        >>> print(td["episode_reward"])
+        tensor([0.])
+        >>> td = env.rollout(3)
+        >>> print(td["next", "episode_reward"])
+        tensor([[-0.5926],
+                [-1.4578],
+                [-2.7885]])
     """
 
     def __init__(
         self,
         in_keys: Optional[Sequence[NestedKey]] = None,
         out_keys: Optional[Sequence[NestedKey]] = None,
+        reset_keys: Optional[Sequence[NestedKey]] = None,
     ):
         """Initialises the transform. Filters out non-reward input keys and defines output keys."""
-        if in_keys is None:
-            in_keys = ["reward"]
-        if out_keys is None and in_keys == ["reward"]:
-            out_keys = ["episode_reward"]
-        elif out_keys is None:
-            raise RuntimeError(
-                "the out_keys must be specified for non-conventional in-keys in RewardSum."
+        # if in_keys is None:
+        #     in_keys = ["reward"]
+        # if out_keys is None:
+        #     out_keys = [
+        #         _replace_last(in_key, f"episode_{_unravel_key_to_tuple(in_key)[-1]}")
+        #         for in_key in in_keys
+        #     ]
+        # if len(in_keys) != len(out_keys):
+        #     raise ValueError(
+        #         "RewardSum expects the same number of input and output keys"
+        #     )
+        # if done_keys is None:
+        #     # The default if done_keys is not provided is that we expect
+        #     # to find the dones in the same tensordicts as the in_keys
+        #     self._done_keys = [_replace_last(in_key, "done") for in_key in in_keys]
+        # elif isinstance(done_keys, list) and len(done_keys) == 1:
+        #     # If there is one done key than it is used for all rewards
+        #     self._done_keys = done_keys * len(in_keys)
+        # elif isinstance(done_keys, list) and len(done_keys) == len(in_keys):
+        #     # The full list of done_keys has been provided
+        #     self._done_keys = done_keys
+        # else:
+        #     raise ValueError(
+        #         f"Provided done_keys should be a list of length 1 or length equals to the number of in_keys, got {done_keys}"
+        #     )
+        #
+        super().__init__(in_keys=in_keys, out_keys=out_keys)
+        self._reset_keys = reset_keys
+
+    @property
+    def in_keys(self):
+        in_keys = self.__dict__.get("_in_keys", None)
+        if in_keys in (None, []):
+            # retrieve rewards from parent env
+            parent = self.parent
+            if parent is None:
+                # fall back on in_keys = ["reward"]
+                in_keys = ["reward"]
```

Comment:
I think "reward" is better

Reply:
sorry I'm not following, what is better?

Reply:
the current one, let's just remove the comments

Reply:
the comment explains the line under :)

Reply:
i mean the ones below